.golangci-kal.yml

```diff
@@ -44,6 +46,12 @@ linters:
           # optionalOrRequired:
           #   preferredOptionalMarker: optional | kubebuilder:validation:Optional # The preferred optional marker to use, fixes will suggest to use this marker. Defaults to `optional`.
           #   preferredRequiredMarker: required | kubebuilder:validation:Required # The preferred required marker to use, fixes will suggest to use this marker. Defaults to `required`.
+            optionalFields:
+              pointers:
+                preference: Always # Always | WhenRequired # Whether to always require pointers, or only when required. Defaults to `Always`.
```

Comment:
IMO, this should be `WhenRequired`, but it will mean more work now.

When set to `WhenRequired`, we will be required to set a minimum length, and min/max values on all string and numeric values respectively.

The linter will then determine whether or not the zero value of the field is valid. If the zero value is not valid, then the optional field does not need to be a pointer, so we can remove it and make the code simpler.

Where the zero value is valid (eg replicas fields, set minimum to 0), then it will report that the field needs to be a pointer still

Reply:
Done